lib/chef/log.rb

```diff
@@ -34,6 +35,14 @@ def self.show_time=(*args)
       end
     end
 
+    def self.deprecation(msg=nil, &block)
+      if Chef::Config[:treat_deprecation_warnings_as_errors]
+        error(msg, &block)
+        raise Chef::Exceptions::DeprecatedFeatureError
```

Comment:
Can we pass the message in here (and handle the case it's an array)? We aggressively turn off logging during spec tests for example, so you need the message in the exception.